contracts/p1/mixins/RewardableLib.sol

```diff
@@ -16,32 +16,33 @@ library RewardableLibP1 {
     using Address for address;
     using SafeERC20 for IERC20;
 
+    event Complete();
+
     // === Used by Traders + RToken ===
 
     /// Claim all rewards
     /// @custom:interaction mostly CEI but see comments
     // actions:
-    //   do asset.delegatecall(abi.encodeWithSignature("claimRewards()")) for asset in assets
+    //   do asset.claimRewards() for asset in assets
     function claimRewards(IAssetRegistry reg) internal {
         Registry memory registry = reg.getRegistry();
-        for (uint256 i = 0; i < registry.assets.length; ++i) {
-            // Claim rewards via delegatecall
-            address(registry.assets[i]).functionDelegateCall(
-                abi.encodeWithSignature("claimRewards()"),
-                "rewards claim failed"
-            );
+        for (uint256 i = 0; i < registry.erc20s.length; ++i) {
+            // empty try/catch because not every erc20 will be wrapped & have a claimRewards func
+            // solhint-disable-next-line
+            try IRewardable(address(registry.erc20s[i])).claimRewards() {} catch {}
         }
+        // weird solidity compiler bug
```

Comment:
wtfff

Reply:
actually turned out to be a weird hardhat-ethers bug, thankfully. reproduced and submitted here: https://github.com/NomicFoundation/hardhat/issues/3880

ill fix this comment